File: app/src/main/java/com/jws/jwsapi/utils/ToastHelper.java
package com.jws.jwsapi.utils;

import android.content.Context;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import com.jws.jwsapi.R;


public class ToastHelper {
    private static Toast currentToast;

    public static void message(String text, int layoutResId, Context context) {
        if (currentToast != null) currentToast.cancel();

        LayoutInflater inflater = LayoutInflater.from(context);
        View layout = inflater.inflate(layoutResId, null);

        TextView textView = layout.findViewById(R.id.text);
        textView.setText(text);
        currentToast = new android.widget.Toast(context);
        currentToast.setGravity(Gravity.CENTER_VERTICAL, 0, 0);
        currentToast.setDuration(android.widget.Toast.LENGTH_LONG);
        currentToast.setView(layout);
        currentToast.show();
    }

}

File: app/src/main/java/com/jws/mobile/core/utils/ToastHelper.kt
package com.jws.mobile.core.utils

import android.content.Context
import android.view.Gravity
import android.view.LayoutInflater
import android.widget.TextView
import android.widget.Toast
import com.jws.mobile.R

object ToastHelper {
    private var currentToast: Toast? = null

    fun message(text: String?, layoutResId: Int, context: Context?) {
        if (currentToast != null) currentToast!!.cancel()

        val inflater = LayoutInflater.from(context)
        val layout = inflater.inflate(layoutResId, null)

        val textView = layout.findViewById<TextView>(R.id.text)
        textView.text = text
        currentToast = Toast(context)
        currentToast!!.setGravity(Gravity.CENTER_VERTICAL, 0, 0)
        currentToast!!.duration = Toast.LENGTH_LONG
        currentToast!!.view = layout
        currentToast!!.show()
    }
}
